### PR TITLE
#168781638 user edit delete comment 

### PR DESCRIPTION
**Pull request by ngireric123**
#### What does this PR do?
- Edit or delete a comment
#### Description of Task to be completed?
- user can delete or edit his/her comment

#### How should this be manually tested?

- Clone this branch
- run npm i
- run npm run dev
- in your browser navigate to localhost:<:port>/requests
- click on comments
- edit or delete comment

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#168781638
#### Screenshots (if appropriate)
image

#### Questions:
N/A

### CODE REVIEW

**Review by alainmateso — APPROVED** [review]
LGTM

**Comment by Cheza-Dzabala** [comment]
Hi @ngireric123 Great job! However please make the following changes:

1. An edited comment should not jump to the top of the pile, it should remain in its original position but may have a small tag that says `edited`

2. Based on the image below, this design is not very intuitive. add an extra `Check Mark` icon so the user clicks when they finish commenting. 

Screen Shot 2020-01-07 at 17 15 32

That aside, great job

File: src/components/pages/requests/CommentDisplay.js [hunk 1]
```diff
@@ -42,23 +67,81 @@ export class CommentDisplay extends Component {
...
+    this.setState({ draftComment: comment, commentId: id }, () => {
+      this.setState({ editMode: true });
+    });
+  }
+
+  sendCommentEdit = async (e) => {
+    e.preventDefault();
+    const { requestId, getComment } = this.props;
+    const { commentId, draftComment } = this.state;
+    const comment = { comment: draftComment };
```
Comment (nakiwuge) [inline src/components/pages/requests/CommentDisplay.js]: Add a new line

File: src/components/pages/requests/CommentDisplay.js [hunk 2]
```diff
@@ -42,23 +67,81 @@ export class CommentDisplay extends Component {
...
+    await deleteComment(commentId);
+    await getComment(requestId);
+    this.setState({
+      isProcessing: false,
+    });
+  }
+
+  editButton = (e) => {
+    e.preventDefault();
+    const { comment, id } = e.target;
```
Comment (nakiwuge) [inline src/components/pages/requests/CommentDisplay.js]: Add a new line

File: src/tests/actionsTests/commentActionTest.spec.js [hunk 3]
```diff
@@ -96,6 +97,79 @@ it('Should trigger GET_COMMENTS', async () => {
...
+    store = mockStore({});
+    await store.dispatch(deleteComment())
+      .then(async () => {
+        const calledActions = store.getActions();
+        expect(calledActions).toEqual(expectedActions);
+      });
+  });
+
+
+
```
Comment (nakiwuge) [inline src/tests/actionsTests/commentActionTest.spec.js]: Remove these lines

File: src/tests/actionsTests/commentActionTest.spec.js [hunk 4]
```diff
@@ -96,6 +97,79 @@ it('Should trigger GET_COMMENTS', async () => {
...
+
+
+
+
+
+
+
```
Comment (nakiwuge) [inline src/tests/actionsTests/commentActionTest.spec.js]: Remove the above lines

File: src/views/requests/SingleRequest.js [hunk 5]
```diff
@@ -172,4 +172,4 @@ SingleRequest.propTypes = {
   dataError: PropTypes.any,
 };
 
-export default connect(mapStateToProps, { singleRequestAction, processRequestAction })(SingleRequest);
+export default connect(mapStateToProps, { singleRequestAction, processRequestAction })(SingleRequest);
```
Comment (nakiwuge) [inline src/views/requests/SingleRequest.js]: Add a line at the end of the file

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@ngireric123 good job however kindly add a loader on  DELETE

**Review by Cheza-Dzabala — APPROVED** [review]

File: src/components/pages/requests/CommentDisplay.js [hunk 6]
```diff
@@ -42,22 +70,88 @@ export class CommentDisplay extends Component {
...
+    });
+    await getComment(requestId);
+    this.setState({
+      isProcessing: false,
+    });
+  }
+
+  editButton = (e) => {
+    e.preventDefault();
+
```
Comment (nakiwuge) [inline src/components/pages/requests/CommentDisplay.js]: Remove this line

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**